pkg/eval/mods/file/file_test.go

```diff
@@ -12,6 +12,12 @@ func TestFile(t *testing.T) {
 		ev.AddGlobal(eval.NsBuilder{}.AddNs("file", Ns).Ns())
 	}
 	evaltest.TestWithSetup(t, setup,
-		evaltest.That(`file:fopen`).Throws(evaltest.AnyError),
+		evaltest.That(`echo haha > out3`, `f = (file:open out3)`, `slurp <$f`, ` file:close $f`).Puts("haha\n"),
+		//No name is specified hence should give an error.
+		evaltest.That(`file:open`).Throws(evaltest.AnyError),
+		evaltest.That(`file:close`).Throws(evaltest.AnyError),
+
+		//Should given an error when try to close an already closed file.
```

Comment:
Space after the `//`.

Also, please squash your commits into a single commit and do a force push. The four commits makes reviewing this much harder. See, for example, https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history but the gist is to execute `git rebase -i HEAD~5` and replace the word "pick" with "s" (or "squash") on every commit but the first one.

Reply:
FYI: Before I squash rebased your commits they added up to 197 insertions and 115 deletions (as reported by `git log --stat -4`). After squashing your commits there are 84 insertions and 2 deletions. The squashed series of commits is considerably more compact and easier to review.

Reply:
This is the second time you are giving me the same feedback, I would like to apologize for my ignorance, but I didn't understand what did you mean by squash push at first. I did look up ```git force push``` but could not find learn and understand anything. Thank you for sending me the link. I will take care of it in future.

Reply:
@supreetsingh10: No need to apologize. I get the sense you are young and inexperienced. Much like the college interns I see at my $day-job. We want you to learn from your experience contributing to this project and be successful at getting changes you authored merged. You simply need to be willing to accept feedback. When you don't understand a suggestion like my earlier  suggestion to do a "squash-merge" you should ask for clarification. For example, "What do you mean by squash-merge?"

Reply:
@supreetsingh10: My recommendation is to squash your commits to a single commit using `git rebase -i HEAD~4` and changing the "pick" to "squash" for the last three commits. Then make the changes @xiaq and I recommended, merge those with `git commit --amend -a`, then force push to Github for review with `git push -f`.

Reply:
Yes, sir, I am always looking forward to the feedback, and at the same time I would not like to burden you with my questions, so I research things myself and patch up the work that has been assigned to me. Nevertheless, since you have made this clear, I will not hesitate to ask questions in the future.